sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,427 @@
+ *
+ * <p>Please refer to the <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/blob-batch">Azure Docs</a>
+ * for more information.</p>
+ */
+public final class BlobBatch {
+    private static final String X_MS_VERSION = "x-ms-version";
+    private static final String BATCH_REQUEST_CONTENT_ID = "Batch-Request-Content-Id";
+    private static final String BATCH_REQUEST_URL_PATH = "Batch-Request-Url-Path";
+    private static final String CONTENT_ID = "Content-Id";
+    private static final String BATCH_BOUNDARY_TEMPLATE = "batch_%s";
+    private static final String REQUEST_CONTENT_TYPE_TEMPLATE = "multipart/mixed; boundary=%s";
+    private static final String BATCH_OPERATION_CONTENT_TYPE = "Content-Type: application/http";
+    private static final String BATCH_OPERATION_CONTENT_TRANSFER_ENCODING = "Content-Transfer-Encoding: binary";
+    private static final String BATCH_OPERATION_CONTENT_ID_TEMPLATE = "Content-ID: %d";
+    private static final String HTTP_VERSION = "HTTP/1.1";
+    private static final String OPERATION_TEMPLATE = "%s %s %s";
+    private static final String HEADER_TEMPLATE = "%s: %s";
+
+    /*
+     * Track the status codes expected for the batching operations here as the batch body does not get parsed in
+     * Azure Core where this information is maintained.
+     */
+    private static final int[] EXPECTED_DELETE_STATUS_CODES = {202};
+    private static final int[] EXPECTED_SET_TIER_STATUS_CODES = {200, 202};
+
+    private final ClientLogger logger = new ClientLogger(BlobBatch.class);
+
+    private final BlobAsyncClient blobAsyncClient;
+
+    private final Deque<Mono<? extends Response<?>>> batchOperationQueue;
+    private final List<ByteBuffer> batchRequest;
+    private final Map<Integer, BlobBatchOperationResponse<?>> batchMapping;
+
+    private final AtomicInteger contentId;
+    private final String batchBoundary;
+    private final String contentType;
+
+    private BlobBatchType batchType;
+
+    BlobBatch(String accountUrl, HttpPipeline pipeline) {
+        this.contentId = new AtomicInteger(0);
+        this.batchBoundary = String.format(BATCH_BOUNDARY_TEMPLATE, UUID.randomUUID());
+        this.contentType = String.format(REQUEST_CONTENT_TYPE_TEMPLATE, batchBoundary);
+
+        boolean batchHeadersPolicySet = false;
+        HttpPipelineBuilder batchPipelineBuilder = new HttpPipelineBuilder().httpClient(this::setupBatchOperation);
+        for (int i = 0; i < pipeline.getPolicyCount(); i++) {
+            HttpPipelinePolicy policy = pipeline.getPolicy(i);
+
+            if (policy instanceof SharedKeyCredentialPolicy) {
+                batchHeadersPolicySet = true;
+                // The batch policy needs to be added before the SharedKey policy to run preparation cleanup.
+                batchPipelineBuilder.policies(this::cleanseHeaders, this::setRequestUrl);
+            }
+
+            batchPipelineBuilder.policies(pipeline.getPolicy(i));
```

Comment:
```suggestion
            batchPipelineBuilder.policies(policy);
```

Reply:
You've already evaluated it in line 99.

Reply:
Will update in Preview 5

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5756